src/vstest.console/Processors/TestAdapterLoadingStrategyArgumentProcessor.cs

```diff
@@ -0,0 +1,276 @@
+    /// The name of the command line argument that the TestAdapterLoadingStrategyArgumentProcessor handles.
+    /// </summary>
+    public const string CommandName = "/TestAdapterLoadingStrategy";
+
+    #endregion
+
+    private Lazy<IArgumentProcessorCapabilities> _metadata;
+
+    private Lazy<IArgumentExecutor> _executor;
+
+    /// <summary>
+    /// Gets the metadata.
+    /// </summary>
+    public Lazy<IArgumentProcessorCapabilities> Metadata
+    {
+        get
+        {
+            if (_metadata == null)
+            {
+                _metadata = new Lazy<IArgumentProcessorCapabilities>(() => new TestAdapterLoadingStrategyArgumentProcessorCapabilities());
+            }
+
+            return _metadata;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the executor.
+    /// </summary>
+    public Lazy<IArgumentExecutor> Executor
+    {
+        get
+        {
+            if (_executor == null)
+            {
+                _executor = new Lazy<IArgumentExecutor>(() => new TestAdapterLoadingStrategyArgumentExecutor(CommandLineOptions.Instance, RunSettingsManager.Instance, ConsoleOutput.Instance, new FileHelper()));
+            }
+
+            return _executor;
+        }
+
+        set
+        {
+            _executor = value;
+        }
+    }
+}
+
+/// <summary>
+/// The argument capabilities.
+/// </summary>
+internal class TestAdapterLoadingStrategyArgumentProcessorCapabilities : BaseArgumentProcessorCapabilities
+{
+    public override string CommandName => TestAdapterLoadingStrategyArgumentProcessor.CommandName;
+
+    public override bool AllowMultiple => false;
+
+    public override bool IsAction => false;
+
+    public override bool AlwaysExecute => true;
+
+    public override ArgumentProcessorPriority Priority => ArgumentProcessorPriority.TestAdapterLoadingStrategy;
+
+    public override string HelpContentResourceName => CommandLineResources.TestAdapterLoadingStrategyHelp;
+
+    public override HelpContentPriority HelpPriority => HelpContentPriority.TestAdapterLoadingStrategyArgumentProcessorHelpPriority;
+}
+
+/// <summary>
+/// The argument executor.
+/// </summary>
+internal class TestAdapterLoadingStrategyArgumentExecutor : IArgumentExecutor
+{
+    #region Fields
+    /// <summary>
+    /// Used for getting sources.
+    /// </summary>
+    private readonly CommandLineOptions _commandLineOptions;
+
+    /// <summary>
+    /// Run settings provider.
+    /// </summary>
+    private readonly IRunSettingsProvider _runSettingsManager;
+
+    /// <summary>
+    /// Used for sending output.
+    /// </summary>
+    private readonly IOutput _output;
+
+    /// <summary>
+    /// For file related operation
+    /// </summary>
+    private readonly IFileHelper _fileHelper;
+
+
+    private static readonly string[] EmptyStringArray =
+#if NET451
+                                                        new string[0];
+#else
+                                                        Array.Empty<string>();
```

Comment:
[cosmetic] Feel free to ignore

```suggestion
        Array.Empty<string>();
```